File: go/leetcode/1207. Unique Number of Occurrences.go
func uniqueOccurrences(arr []int) bool {
    if len(arr) == 1 {return true}
    mp := make(map[int]int)
    mp1 := make(map[int]int)
    
   for _, val := range arr {
        v, ok := mp[val]
        if ok {
            mp[val] = v + 1
        } else {
            mp[val] = 1
        }
    }
    
    
   for _, val := range arr {
        v, ok := mp[val]
        if ok {
            mp1[v] = 0
        }
    } 
    
    return len(mp1) == len(mp)
}

File: javascript/leetcode/1207. Unique Number of Occurrences.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function(arr) {
  if (arr.length === 1) return arr;

  const map = new Map();

  arr.forEach(el => {
    if (map.has(el)) {
      map.set(el, map.get(el) + 1);
    } else map.set(el, 1);
  });
  return new Set(map.values()).size === map.size;
};
